### PR TITLE
[lambda] Enforce mutual exclusivity between responseTransformation and unwrapAs

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

This PR fixes bug whereby setting `responseTransformation: true` alongside `unwrapAsApiGateway: true` or `unwrapAsAlb: true` would set a transformation that overwrote the header `Content-Type: text/html` regardless of the metadata included in the lambda response.

### Note
It appears to be the case that the `responseTransformation` feature is a precursor to the more complete `unwrapAs*` features, and likewise for `requestTransformation` and `wrapAsApiGateway`. If this is the case, it seems to makes sense to deprecate `responseTransformation` and `requestTransformation` and simplify the plugin code by treating them equivalently

# Testing steps
Build test chart, images, push to kind:
```shell
make build-test-chart docker push-kind-images
```

Install to clean kind cluster (use the chart name built by previous command)
```shell
helm install --create-namespace gloo _test/gloo-1.14.0-beta9-10-g38c620b7e.tgz -n gloo-system
```

Create AWS creds secret
```shell
glooctl create secret aws --name 'aws-creds' --namespace gloo-system --access-key $ACCESS_KEY --secret-key $SECRET_KEY
```

Create upstream
```shell
glooctl create upstream aws \
       --name 'aws-upstream' \
       --namespace 'gloo-system' \
       --aws-region 'us-east-1' \
       --aws-secret-name 'aws-creds' \
       --aws-secret-namespace 'gloo-system'
```

Create VirtualService
```shell
kubectl apply -f - <<EOF
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: aws-route
  namespace: gloo-system
spec:
  virtualHost:
    domains:
    - '*'
    routes:
    - matchers:
      - exact: '/'
      routeAction:
        single:
          destinationSpec:
            aws:
              logicalName: test-unwrap-json
              transformResponse: false
              unwrapAsAlb: false
        upstream:
            name: aws-upstream
            namespace: gloo-system
EOF
```

Toggle on/off the `transformResponse` and `unwrapAsAlb` to see the content type header is overridden to `text/html` iff `transformResponse: true` and `unwrapAsAlb: false`

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: changelog/v1.14.0-beta10/lambda-unwrap-no-transform.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+changelog:
+- type: FIX
+  issueLink: https://github.com/solo-io/solo-projects/issues/4434
+  resolvesIssue: false
+  description: >-
+    Do not add transformations to lambda response when we are unwrapping since those filters handle the response in its entirety
```
Comment (nfuden) [inline changelog/v1.14.0-beta10/lambda-unwrap-no-transform.yaml:6]: nit: Lets make this more user facing. Namely why should they care what is the output affect to the user/

**Comment by nfuden** [comment]
Are we saying that no transformations are compatible with unwrap? Shouldnt there be a way to respect transformations before/after unwrap?

**Comment by jbohanon** [comment]
> Are we saying that no transformations are compatible with unwrap? Shouldnt there be a way to respect transformations before/after unwrap?

No, we're saying the setting `responseTransformation` on the [AWS destination spec](https://docs.solo.io/gloo-edge/latest/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options/aws/aws.proto.sk/#destinationspec) is incompatible with `unwrapAsApiGateway` and `unwrapAsAlb` since it will [hard-code the content-type header](https://github.com/solo-io/gloo/blob/82316ac89210e49554c7fec2fb1afe1afe10ab99/projects/gloo/pkg/plugins/aws/plugin.go#L263-L280) ignoring the metadata in the response body.

I'm not sure what opinions we do or should have about them setting other transformations, but this automatic one we inject mangles the expected response

File: changelog/v1.14.0-beta10/lambda-unwrap-no-transform.yaml (hunk 1 above)
Comment (ben-taussig-solo) [reply changelog/v1.14.0-beta10/lambda-unwrap-no-transform.yaml:6]: Maybe something like
 > Makes requestTransformation and unwrapAsAlb/unwrapAsApiGateway Lambda DestinationSpec settings mutually exclusive

**Comment by jbohanon** [comment]
/kick [Install [It] shouldn't get errors for knative dry run](https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=ac436797-cad0-4f91-a4b7-636ae4de95d1)

**Review by ben-taussig-solo — APPROVED** [review]

**Review by nfuden — APPROVED** [review]

**Pull request merged**